code/modules/mob/mob.dm

```diff
@@ -1271,3 +1271,34 @@
 /mob/proc/active_storage_deleted(datum/source)
 	SIGNAL_HANDLER
 	set_active_storage(null)
+
+/// if you hear a radio talk that you don't have access to for some reason, you can't recognise its channel name. This updates your known channel depending on radios in your possession
+/mob/proc/refresh_known_radio_channels()
+	known_channels = list() // reset first
+	for(var/obj/item/radio/mob_radio in src.get_contents())
+		if(mob_radio.listening) // if not turned off, you shouldn't be eligible for this.
+			if(istype(mob_radio, /obj/item/radio/headset)) // only headset has 2nd keyslot
+				if(!iscarbon(src)) // if a non-carbon somehow has a headset, it will cause runtime.
+					stack_trace("a non-carbon mob has a headset [mob_radio].") // If a mob is fine to have headset, remove this runtime, nor a mob having headset is wrong.
+					continue
+				var/mob/living/carbon/carbon = src
+				var/obj/item/radio/headset/headset_on_ears = mob_radio // only headset has `keyslot2` variable
+				if(headset_on_ears == carbon.ears) // this must be worn by you, not in your inventory.
+					known_channels["[headset_on_ears.frequency]"] = TRUE
+					for(var/K in headset_on_ears.keyslot?.channels)
+						known_channels["[GLOB.radiochannels[K]]"] = TRUE
+					for(var/K in headset_on_ears.keyslot2?.channels)
+						known_channels["[GLOB.radiochannels[K]]"] = TRUE
+			else
+				known_channels["[mob_radio.frequency]"] = TRUE
+				for(var/K in mob_radio.keyslot?.channels)
+					known_channels["[GLOB.radiochannels[K]]"] = TRUE
```

Comment:
Not sure how difficult it would be to implement, but can I suggest a far, far better approach?
Store a dictionary on the mob with a list of channels that they know and how many times they know them.
When a headset is equipped to the ear slot, increment the counter by 1.
When a headset is unequipped **from the ear slot**, increment the counter by 1.
For radios do the same, but have it when added to any slot on the mob.

This would completely avoid looping over everything on the mob (get_contents() can be real slow) and would make this not require a loop at all. The only challenge is ensuring consistency when decrementing and incrementing

Reply:
![image](https://user-images.githubusercontent.com/87972842/230516760-ba1482cc-7fa0-40b8-bacf-064522ac48a7.png)

this is why it's not possible. If you move a thing within in your hand, it calls 'pick()' verb every time.